### PR TITLE
Issue #40 - Type error in LambdaUtils and in example

### PR DESCRIPTION
**Pull request by kernwig**
- The APIGatewayEvent type is deprecated and replaced with APIGatewayProxyEvent
- Update AsyncProxyHandler to expect the new LambdaUtils version of APIGatewayProxyEvent.
- Fix bugs in documentation's example for LambdaUtils

### CODE REVIEW

File: docs/source/lambda_utils.rst [hunk 1]
```diff
@@ -72,7 +72,7 @@ General use
     export const hello = LambdaUtils.wrapApiHandler(async (event: LambdaUtils.APIGatewayProxyEvent) => {
         // These event objects are now always defined, so don't need to check for undefined. 🙂
         const who = event.pathParameters.who;
-        const points = parseInt(event.queryStringParameters.points || 0);
+        let points = Number(event.queryStringParameters.points || '0');
```
Comment (ShaunEdiger) [inline docs/source/lambda_utils.rst:75]: Curiosity: why the change from `const` to `let`?  (Was this the source of the error?)

**Comment by kernwig** [comment]
Fixes issue #46
